src/main/java/core/basesyntax/services/DataConverterImpl.java

```diff
@@ -0,0 +1,40 @@
+package core.basesyntax.services;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.ArrayList;
+import java.util.List;
+
+public class DataConverterImpl implements DataConverter {
+    @Override
+    public List<FruitTransaction> convertToTransaction(List<String> data) {
+        List<FruitTransaction> transactions = new ArrayList<>();
+
+        for (int i = 1; i < data.size(); i++) {
+            String line = data.get(i);
+            String[] parts = line.split(",");
+
+            if (parts.length < 3) {
+                throw new IllegalArgumentException("Invalid data format at line "
+                        + (i + 1) + ": " + line);

```

Comment:
The error message 'Invalid data format' could be more informative by specifying what the expected format is. For example, you could include the expected number of parts in the message.